core/src/main/kotlin/com/github/quillraven/darkmatter/screen/MenuScreen.kt

```diff
@@ -1,11 +1,115 @@
 package com.github.quillraven.darkmatter.screen
 
 import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.utils.Align
 import com.github.quillraven.darkmatter.Game
+import com.github.quillraven.darkmatter.asset.I18NBundleAsset
 import com.github.quillraven.darkmatter.asset.MusicAsset
+import com.github.quillraven.darkmatter.ecs.createDarkMatter
+import com.github.quillraven.darkmatter.ecs.createPlayer
+import com.github.quillraven.darkmatter.ecs.system.MoveSystem
+import com.github.quillraven.darkmatter.ecs.system.PlayerAnimationSystem
+import com.github.quillraven.darkmatter.ecs.system.PowerUpSystem
+import com.github.quillraven.darkmatter.ui.SkinImageButton
+import com.github.quillraven.darkmatter.ui.SkinLabel
+import com.github.quillraven.darkmatter.ui.SkinTextButton
+import ktx.actors.onChangeEvent
+import ktx.actors.onClick
+import ktx.ashley.getSystem
+import ktx.scene2d.Scene2dDsl
+import ktx.scene2d.StageWidget
+import ktx.scene2d.actors
+import ktx.scene2d.imageButton
+import ktx.scene2d.label
+import ktx.scene2d.table
+import ktx.scene2d.textButton
+
+private const val PLAYER_SPAWN_Y = 3f
+private const val OFFSET_TITLE_Y = 15f
+private const val MENU_ELEMENT_OFFSET_TITLE_Y = 20f
+private const val MENU_DEFAULT_PADDING = 5f
 
 class MenuScreen(game: Game) : Screen(game, MusicAsset.MENU) {
+    private val bundle = assets[I18NBundleAsset.DEFAULT.descriptor]
+    private val highScoreText = "${bundle["highscore"]}: "
+
+    override fun show() {
+        super.show()
+        engine.run {
+            getSystem<PowerUpSystem>().setProcessing(false)
+            getSystem<MoveSystem>().setProcessing(false)
+            getSystem<PlayerAnimationSystem>().setProcessing(false)
+            createPlayer(assets, spawnY = PLAYER_SPAWN_Y)
+            createDarkMatter()
+        }
+
+        stage.actors {
+            createMenuUI()
+        }
+    }
+
+    private fun @Scene2dDsl StageWidget.createMenuUI() {
```

Comment:
This might be less confusing for new KTX users as a regular method rather than an extension function.

```kotlin
private fun createMenuUI(stage: Stage) {
  stage.actors {
    table {
      // (...)
    }
  }
}
```

Reply:
Normally you do not have to use the `@Scene2dDsl` annotation as a user.